File: 1669. Merge In Between Linked Lists/Solution.java
public class ListNode {
    int val;
    ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode p = list1;
        for (int i = 0; i < a - 1; i++)
            p = p.next;
        ListNode q = p;
        for (int i = a - 1; i <= b; i++)
            q = q.next;
        p.next = list2;
        while (p.next != null)
            p = p.next;
        p.next = q;
        return list1;
    }
}

File: 1669. Merge In Between Linked Lists/Solution.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        i, p = 0, list1
        while i < a - 1:
            p = p.next
            i += 1
        q = p
        while i <= b:
            q = q.next
            i += 1
        p.next = list2
        while p.next:
            p = p.next
        p.next = q
        return list1
